packages/chain-bridge/src/router/BridgeRouter.sol

```diff
@@ -404,6 +423,9 @@ contract BridgeRouter is
             options
         );
 
+        // Refund any excess native fees to designated refund address
+        _refund(params.refundAddress, msg.value - bufferedFee);
```

Comment:
is the quoted fee the same as the quoted fee inside of the adapter ? if yes shouldnt the refunds be handled in one place rather than twice ?

Reply:
ok but then we do the quoting 3 times:
1. keeper estimates the fee using dummy data
2. fee is estimated using dummy data ( + buffer) in the router
3. fee is quoted in adapter

why not 
1. estimate fee offchain ( dummy data )
2. add buffer to that fee in router ( no estiamte )
3. fee is quoted in adapter

or 
1. quote using real message data + add buffer offchain / add buffer in the estimate method
2. don't add buffer in router
3. refund in adapter